pubsub/gcppubsub/gcppubsub.go

```diff
@@ -212,13 +216,12 @@ func (o *URLOpener) OpenTopicURL(ctx context.Context, u *url.URL) (*pubsub.Topic
 // OpenSubscriptionURL opens a pubsub.Subscription based on u.
 func (o *URLOpener) OpenSubscriptionURL(ctx context.Context, u *url.URL) (*pubsub.Subscription, error) {
 	// Set subscription options to use defaults
-	o.SubscriptionOptions = SubscriptionOptions{
-		MaxBatchSize: defaultRecvBatcherOpts.MaxBatchSize,
-	}
+	opts := o.SubscriptionOptions
+	opts.MaxBatchSize = defaultRecvBatcherOpts.MaxBatchSize
```

Comment:
Shouldn't need this (and in fact don't want it, in case there is a custom `URLOpener` with it set we don't want to overwrite it). If the value is 0, it will get the default via openSubscriber later.